src/ipc/mem/Segment.cc

```diff
@@ -277,8 +278,7 @@ Ipc::Mem::Segment::GenerateName(const char *id)
         if (name[name.size()-1] != '/')
             name.append('/');
     } else {
-        name.append('/');
-        name.append(service_name.c_str());
+        name.append(Instance::NamePrefix("/").c_str());
```

Comment:
`c_str()` performs a data copy, which is bad when we are already forced to do a data-copy to get the result into `String`.
```suggestion
        const auto prefix = Instance::NamePrefix("/");
        name.append(prefix.rawContent(), prefix.length());
```

Reply:
Yes, `SBuf::c_str()` should be avoided whenever possible.

I disagree with your "not good enough" - it works, does not add new bugs, improves the code, and meets the scope of this PR. IMHO, further improvement requiring larger rewrites is perfectionism that risks going out of scope and delays this bug fix which we need to get merged.

Reply:
> Yes, `SBuf::c_str()` should be avoided whenever possible.

I disagree with the "possible" bit, but I will work on eliminating these c_str() calls from PR changes to make progress.

> I disagree with your "not good enough" - it works, does not add new bugs, improves the code, and meets the scope of this PR. IMHO, further improvement requiring larger rewrites is perfectionism that risks going out of scope and delays this bug fix which we need to get merged.

I disagree that the proposed replacement improves code. I have already detailed the reasons behind that assessment.

The problem this change request is targeting is indeed out of this PR scope. Insisting on a solution delays this PR.

Reply:
Done